snapm/fsdiff/cache.py

```
@@ -238,14 +318,78 @@ def _valid(timestamp: int, expires: int = _CACHE_EXPIRES_SECS) -> bool:
+            try:
+                progress.start(candidate.count)
+            except AttributeError as attr_err:
+                _log_debug(
+                    "Ignoring mistmatched cache file version: %s",
+                    attr_err,
+                )
+                return None
+            for i in range(0, candidate.count):
+                progress.progress(i, f"Loading record {i}")
+                try:
+                    record = pickle.load(reader)
```

Ruff (0.14.8)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)